[WPE] Modern media controls cause timeouts in layout tests

aperezdc opened the pull request:
#### 4cd7ac02f6bd9a07272753ae40ae21f0bdf7cac8
<pre>
[WPE] Modern media controls cause timeouts in layout tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=244378">https://bugs.webkit.org/show_bug.cgi?id=244378</a>

Reviewed by Michael Catanzaro.

* Source/WebCore/CMakeLists.txt: Arrange adding one more CSS file for
  modern media controls when building the GTK or the WPE port.
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-overrides.css:
  Added, provides a rule to disable -webkit-backdrop-filter for some
  elements.
* Tools/TestWebKitAPI/glib/TestExpectations.json: Reverts the change
  from 252659@main which marked test case
  /webkit/WebKitWebView/external-audio-rendering as slow, as it should
  not time out anymore without the backdrop filter effects.

Canonical link: <a href="https://commits.webkit.org/253982@main">https://commits.webkit.org/253982@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/10aa5c4a8b3b311e353b53b77697388bcd1c14ac

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87719 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31805 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/18455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96914 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/150725 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91688 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/30163 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/26264 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79817 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91660 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93338 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/24364 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74466 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/24362 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79326 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/18455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67211 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27857 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/18455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27823 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/18455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2803 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29518 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/37245 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/29416 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/18455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebCore/CMakeLists.txt:
```diff
@@ -2147,6 +2147,12 @@ set(MODERN_MEDIA_CONTROLS_STYLE_SHEETS
     "${WEBCORE_DIR}/Modules/modern-media-controls/controls/watchos-media-controls.css"
 )
 
+if (PORT STREQUAL "GTK" OR PORT STREQUAL "WPE")
```
We would normally duplicate this code between PlatformGTK.cmake and PlatformWPE.cmake, to keep it out of the cross-platform CMakeLists.txt.

aperezdc replied:
Or put it in `PlatformGLib.cmake`, but in this case we need to have the final value of `MODERN_MEDIA_CONTROLS_STYLESHEET` ready before the `add_custom_command()` that defines the target for generating `ModernMediaControls.css` right below... but `PlatformXXX.cmake` gets included by the build system _after_ the `CMakeLists.txt` file was processes, so I'm afraid we don't have a better way in this case.

aperezdc committed:
[WPE] Modern media controls cause timeouts in layout tests
https://bugs.webkit.org/show_bug.cgi?id=244378

Reviewed by Michael Catanzaro.

* Source/WebCore/CMakeLists.txt: Arrange adding one more CSS file for
  modern media controls when building the GTK or the WPE port.
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-overrides.css:
  Added, provides a rule to disable -webkit-backdrop-filter for some
  elements.
* Tools/TestWebKitAPI/glib/TestExpectations.json: Reverts the change
  from 252659@main which marked test case
  /webkit/WebKitWebView/external-audio-rendering as slow, as it should
  not time out anymore without the backdrop filter effects.

Canonical link: https://commits.webkit.org/253982@main
